extra/query_propagation.rst

```diff
@@ -0,0 +1,123 @@
+
+However the way queries are partitioned into fragments (and which queries are propagated at all) varies by the type of query. In some advanced situations it is useful to manually control this behavior. Citus provides utility functions to propagate SQL to workers, shards, or placements.
+
+Running on all Workers
+----------------------
+
+The least granular level of execution is broadcasting a statement for execution on all workers. This is useful for viewing a properties of entire worker databases or creating UDFs uniformly throughout the cluster. For example:
+
+.. code-block:: postgresql
+
+  -- Make a UDF available on all workers
+  SELECT run_command_on_workers($cmd$ CREATE FUNCTION ... $cmd$);
+
+  -- List the work_mem setting of each worker database
+  SELECT run_command_on_workers($cmd$ SHOW work_mem; $cmd$);
+
+.. note::
+
+  The :code:`run_command_on_workers` function and other manual propagation commands in this section can run only queries which return a single column and single row.
+
+Running on all Shards
+---------------------
+
+Each distributed table :code:`foo` has shards across the workers which are each tables with a naming convention of :code:`foo_n` where :code:`n` is a number. In order to run a query against a shard, the query must be run on its containing worker, and the shard name must be interpolated into the query string as an identifier (:code:`%I`).
+
+.. code-block:: postgresql
+
+  -- Get the estimated row count for a distributed table by summing the
+  -- estimated counts of rows for each shard.
+  SELECT sum(result::bigint) AS estimated_count
+    FROM run_command_on_shards(
+      'my_distributed_table',
+      $cmd$
+        SELECT reltuples
+          FROM pg_class
+         WHERE relname = '%I';
+      $cmd$
+    );
+
+Running on all Placements
+-------------------------
+
+When using Citus- rather than streaming-replication for :ref:`dealing_with_node_failures`, each shard has replicas stored as tables on other workers. Shards and their replicas are all called *placements*. (PostgreSQL streaming replication, on the other hand, works at the database level rather than the shard level, so the notion of shard replica placements is not applicable there.)
+
+Queries that update rows ought to be run on all placements rather than simply all shards. Ordinarily update queries go through the coordinator node which ensures they apply across placements. The functions in this section bypass the normal coordinator logic, so skipping any replicas in a manual update will leave the cluster inconsistent. Conversely, read-only queries should be run on shards rather than placements or they will return duplicate results.
+
+The following are equivalent:
+
+.. code-block:: postgresql
+
+  -- ordinary query going through the coordinator
```

Comment:
We don't support this query as-is going through the co-ordinator. We support this via master_modify_multiple_shards. So let's take out this as an equivalent query.